lib/plugins/provisioning/src/test/plans-creation-test.js

```diff
@@ -134,31 +138,91 @@ const testPostOfPlan = (planType, plan, db) => {
   it(`errors on create of already existing ${planType} plan`, (done) => {
     yieldable.functioncb(function *() {
       const id = ['k', plan.plan_id].join('/');
-      yield db.put(extend({}, plan, {
-        _id: id
-      }));
+      yield db.put(extend({}, plan, { _id: id }));
     })((error) => {
       if(error)
         throw error;
-      postRequest(planType, plan, (val) => {
+      postRequest(headers, planType, plan, (val) => {
         expect(val.statusCode).to.equal(409);
         done();
       });
     });
   });
 
   it(`validate post of empty ${planType} plan`, (done) => {
-    postRequest(planType, {}, (val) => {
+    postRequest(headers, planType, {}, (val) => {
       expect(val.statusCode).to.equal(400);
       done();
     });
   });
 
 };
 
-describe('abacus-ext-provisioning-plugin plans creation metering',
-  () => testPostOfPlan('metering', meteringPlan, meteringdb));
-describe('abacus-ext-provisioning-plugin plans creation pricing',
-  () => testPostOfPlan('pricing', pricingPlan, pricingdb));
-describe('abacus-ext-provisioning-plugin plans creation metering',
-  () => testPostOfPlan('rating', ratingPlan, ratingdb));
+describe('When not secured', () => {
+
+  setupDbConnections();
```

Comment:
We should avoid duplication in describe-s. What about extracting the common code in a function that accepts secured flag?